src/ln/msgs.rs

```diff
@@ -43,6 +44,12 @@ pub enum DecodeError {
 	/// A length descriptor in the packet didn't describe the later data correctly
 	/// (currently only generated in node_announcement)
 	BadLengthDescriptor,
+	/// Error from std::io
+	Io(::std::io::Error),
+	/// Invalid value found when decoding
```

Comment:
Can you be more specific here? AFAICT this is only used for bool decoding  (making it unused currently, presumably until we move the ChannelMonitor deserializer to the new framework) when the value is not 1 or 0.